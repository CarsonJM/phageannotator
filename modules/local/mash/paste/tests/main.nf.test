nextflow_process {

    name "Test Process: MASH_PASTE"
    script "../main.nf"
    process "MASH_PASTE"
    tag "modules"
    tag "modules_local"
    tag "mash_paste"


    test("['modules_local']['mash_sketch1_msh'] & ['modules_local']['mash_sketch2_msh']") {

        when {
            params {
                outdir              = "$outputDir"
            }
            process {
                """
                input[0] = [ [id: 'test1'], [file(params.test_data['modules_local']['mash_sketch1_msh'], checkIfExists: true)] ]
                input[1] = [ [id: 'test2'], [file(params.test_data['modules_local']['mash_sketch2_msh'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("['modules_local']['mash_sketch1_msh'] & ['modules_local']['mash_sketch2_msh'] - stub") {

        options "-stub"

        when {
            params {
                outdir              = "$outputDir"
            }
            process {
                """
                input[0] = [ [id: 'test1'], [file(params.test_data['modules_local']['mash_sketch1_msh'], checkIfExists: true)] ]
                input[1] = [ [id: 'test2'], [file(params.test_data['modules_local']['mash_sketch2_msh'], checkIfExists: true)] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
