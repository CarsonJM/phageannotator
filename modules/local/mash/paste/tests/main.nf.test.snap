{
    "['modules_local']['mash_sketch1_msh'] & ['modules_local']['mash_sketch2_msh']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,37a3fadd36ab7a39f099465da3e17430"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ],
                "msh": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,37a3fadd36ab7a39f099465da3e17430"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ]
            }
        ],
        "timestamp": "2023-11-13T11:52:25.887887449"
    },
    "['modules_local']['mash_sketch1_msh'] & ['modules_local']['mash_sketch2_msh'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ],
                "msh": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:33:24.750945115"
    }
}